lumberjack.go

```diff
@@ -356,6 +396,37 @@ func (l *Logger) oldLogFiles() ([]logInfo, error) {
 	return logFiles, nil
 }
 
+// compressLog compresses the log with given filename using Gzip compression
+func compressLog(filename string) error {
+
+	reader, err := os.Open(filename)
```

Comment:
This should probably not be called reader.  Input or something would be better. Reader I generally reserve only for actual io.Reader values.